agent/drive_handler.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type drive struct {
+	Name       string
+	DriveID    string
+	MajorMinor string
+	Size       int
+	Type       string
+}
+
+type driveHandler struct {
+	drives []drive
+}
+
+// DiscoverDrives will use lsblk to populate the drives field in
+// the drive handler.
+func (d *driveHandler) DiscoverDrives() error {
+	names, err := getListOfBlockDeviceNames()
+	if err != nil {
+		return err
+	}
+
+	drives := []drive{}
+	for _, name := range names {
+		d, err := buildDrive(name)
+		if err != nil {
+			return err
+		}
+
+		drives = append(drives, d)
+	}
+
+	for i := 0; i < len(drives); i++ {
+		if !isStubDrive(drives[i]) {
+			continue
+		}
+
+		id := getDriveID(drives[i])
+		drives[i].DriveID = id
+	}
+
+	fmt.Println(drives)
+	d.drives = drives
+	return nil
+}
+
+func getDriveID(d drive) string {
+	return ""
+}
+
+// isStubDrive will check to see if a given drive is a stub drive. This is done
+// by looking to see the size of the drive is exactly 512 bytes
+func isStubDrive(d drive) bool {
+	if d.Size != 512 {
+		return false
+	}
+
+	return true
+}
+
+const (
+	blockPath = "/sys/block"
+
+	blockMajorMinor = "dev"
+	blockSize       = "size"
+)
+
+func getListOfBlockDeviceNames() ([]string, error) {
+	names := []string{}
+	err := filepath.Walk(blockPath, func(path string, info os.FileInfo, err error) error {
```

Comment:
nit: I think all we need here is a single [ReadDir](https://golang.org/pkg/io/ioutil/#ReadDir), nothing recursive